momentGW/energy.py

```diff
@@ -8,7 +8,7 @@
 
 def hartree_fock(rdm1, fock, h1e):
     """Hartree--Fock energy functional."""
-    return lib.einsum("ij,ji->", rdm1, h1e + fock) * 0.5
+    return np.sum(np.multiply(rdm1, h1e + fock)) * 0.5
```

Comment:
Use `einsum` -- it's much clearer. We only need to worry about the optimisation of rate-limiting calls.